### ISSUE TITLE
Unexpected behaviour of BehaviourSubject

### ISSUE THREAD

**Issue by splangi** [ReactiveX/rxdart#236]
In my opinion, the BehaviourSubject is not working correctly, when null's are passed to the add function. The way that BehaviourSubject is implemented in RxDart, null value is ignored as seed, but they are valid to send to the add function. This previous even might be expected, but what's not expected is that after re-listening to a BehaviourSubject, that has multiple values already added to it, still ignores the latest value, if it is null. 

An example: 

`BehaviorSubject<int> subject = BehaviorSubject(seedValue: 0, sync: true);
subject.add(1);
  var subscription = subject.listen((item) => print("subscription1 : " + item.toString()));
  subject.add(null);
  var subscription2 = subject.listen((item) => print("subscription2 : " + item.toString()));
  subject.add(2);`

Expected output:
>flutter: subscription2 : 3
>flutter: subscription1 : 2
>flutter: subscription2 : null
>flutter: subscription1 : null
>flutter: subscription1 : 3

Real output:
>flutter: subscription2 : 3
>flutter: subscription1 : 2
>flutter: subscription1 : null
>flutter: subscription1 : 3`

The code that results in such a behaviour is here:

` return new BehaviorSubject<T>._(
        controller,
        new Observable<T>.defer(
            () => wrapper.latestValue == null
                ? controller.stream
                : new Observable<T>(controller.stream)
                    .startWith(wrapper.latestValue),
            reusable: true),
        wrapper);`

When even re-listened to and the last value was null, then it ignores it, which in my opinion is unexpected to the developer. Either disallow nulls alltogether, make the seedValue mandatory or point out that nulls are semi-supported in the documentation.

**Comment by brianegan** [comment]
Thanks! This is a dupe of #170 -- I'm currently reviewing the fix offered in #195

**Comment by yogesh-aggarwal** [comment]
For anyone coming in future here, the `seedValue` argument is changed to `.seeded(value)` function.

**Instead of:**
```dart
var sub = new BehaviourSubject<int>(seedValue: 2)
```

**Do:**
```dart
var sub = new BehaviourSubject<int>.seeded(2)
```
